### ISSUE TITLE
parsing of authors in CRAN DESCRIPTION files is incorrect

### ISSUE THREAD

**Issue by cboettig** [Impactstory/citeas-api#43]
Consider: http://citeas.org/cite/CRAN.R-project.org/package=arkdb
Note that this fails to capture first author listed in DESCRIPTION. 

This seems relatively common, e.g. http://citeas.org/cite/CRAN.R-project.org/package=pmc doesn't find any author.

**Comment by caseydm** [comment]
I made some updates to improve this. Please take a look!

**Comment by cboettig** [comment]
Great!  Looks much better.  However, this still does not seem to be paying attention the the role codes designated in the citation file.  The citation should only list as authors those with the role code `aut` -- this matches the behavior of `citation()` in R.  (or `cre`, which is implicitly author and maintainer).  

Note that R is using (a subset of) Library Of Congress MARC codes to define roles (https://www.loc.gov/marc/relators/relaterm.html), which can include roles such as `rev` (reviewer), `cph` (copyright holder) which are very distinct from authors.  Also, I believe that being able to distinguish between acknowledging all contributors as such without attaching the additional significance usually reserved the academic notion of 'author'  is sometimes important.

**Comment by caseydm** [comment]
Yes you are right. It's not parsing those role codes. I'll add that in and test it so only roles with `aut` or `cre `are considered as authors in the citation. Somebody else brought this up recently so hopefully this solves the issue they were talking about as well.

**Comment by caseydm** [comment]
Ok, just implemented this feature. An author is only added if associated role is aut or cre.
